app/assets/javascripts/components/general_button.es6.jsx

```diff
@@ -0,0 +1,58 @@
+class GeneralButton extends React.Component {
+
+  static get propTypes() {
+    return {
+      action: React.PropTypes.func,
+      route: React.PropTypes.string,
+      content: React.PropTypes.string.isRequired,
+    };
+  }
+
+  static get defaultProps() {
+    return {
+      action: null,
+      route: '',
+      content: '',
+    };
+  }
+
+  get styles() {
+    return {
+      container: {
+        display: 'inline-block',
+        verticalAlgin: 'middle',
+        position: 'relative',
+        padding: '6px 12px',
+        color: 'white',
+        backgroundColor: '#68B1DE',
+        borderBottom: '2px solid',
+        borderColor: '#28719E',
+        borderRadius: 2,
+        fontSize: '14px',
+      },
+    };
+  }
+
+  handleClick(event) {
+    if (this.props.route === '') {
+      event.preventDefault();
+      event.stopPropagation();
+      if (this.prop.action !== null) {
+        this.props.action();
+      }
+    }
+  }
+
+  render () {
+    console.log(this.props.route);
+    return (
+      <a
+        href={this.props.route}
+        onClick={this.handleClick.bind(this)}
```

Comment:
We set the onClick attribute to the 'this.handleClick' function. What does the '.bind(this)' do?